src/node/genesis_gen.h

```diff
@@ -398,10 +398,41 @@ namespace ccf
       shares_view->put(0, key_share_info);
     }
 
-    void set_recovery_threshold(size_t threshold)
+    bool set_recovery_threshold(size_t threshold)
     {
       auto config_view = tx.get_view(tables.config);
+
+      // While waiting for recovery shares, the recovery threshold cannot be
+      // modified. Otherwise, the threshold could be passed without triggering
+      // the end of recovery procedure
+      if (
+        get_service_status().value() ==
+        ServiceStatus::WAITING_FOR_RECOVERY_SHARES)
+      {
+        LOG_FAIL_FMT(
+          "Failed to set recovery threshold: service is currently waiting for "
+          "recovery shares");
+        return false;
+      }
+
+      if (threshold == 0)
+      {
+        LOG_FAIL_FMT("Recovery threshold cannot be set to 0");
+        return false;
+      }
+
+      auto active_members_count = get_active_members_count();
+      if (threshold > active_members_count)
+      {
+        LOG_FAIL_FMT(
+          "Recovery threshold cannot be set to {} as it is greater than "
+          "the number of active members ({})",
```

Comment:
Minor: Is it worth enforcing this at this level? AFAICT, we don't do the same test in the opposite direction (ie - retiring members can move you below the recovery threshold). Can we reliably prevent members from putting the network in an unrecoverable state? If we can't, should we be suggesting (with errors like this) that we _are_ generally enforcing this?

Reply:
I think we ought to try to validation on retiring members as well. Putting the network in an irrecoverable state is such a far-reaching decision that it should be a clearly explicit "make_network_irrecoverable" proposal if it's allowed at all, not something you can sleepwalk into.

Reply:
Good spot! I moved this code from `member_frontend.h` but did not move the opposite code to here. Just to clarify: we already do enforce that we never end up in a non-recoverable situation by checking that the number of active members is always `>=` than the recovery threshold.